### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by kkuzmin**
### Problem Description

The current implementation is more AWS specific.

### Solution Description

Make it more Azure friendly.

### CODE REVIEW

File: al_servicec.js [hunk 1]
```diff
@@ -164,19 +173,33 @@ class IngestC extends AlServiceC {
...
+                'x-invoked-by' : this._functionType,
                 'Content-Encoding' : 'deflate',
                 'Content-Length' : Buffer.byteLength(data)
             },
             body : data
         };
         return this.post(`/data/vpcflow`, payload);
     }
+    
+    sendAicspmsgs(data, invokedBy) {
```
Comment (tomdos) [inline al_servicec.js]: Just out of curiosity what does it stand for `Aicsp` ? Maybe more camel case here ;-)?

Comment (kkuzmin) [reply al_servicec.js]: it stands for "appliance in cloud send parsed" -  I don't think we need more camels here =)

File: al_servicec.js [hunk 2]
```diff
@@ -208,21 +261,72 @@ class AzcollectC extends AlServiceC {
...
+    register(registrationValues) {
+        switch(this._collectorType) {
+            case COLLECTOR_TYPES.CWE:
+            case COLLECTOR_TYPES.CWL:
+                return this._doRegistrationAws(registrationValues);
+                break;
+            case COLLECTOR_TYPES.O365:
+                return this._doRegistrationAzure(registrationValues);
+            default:
+                break;
```
Comment (tomdos) [inline al_servicec.js]: I would suggest to throw an exception for default case in register/deregister/checkin. This should not happen so let's not process it silently

Comment (kkuzmin) [reply al_servicec.js]: This exception will be thrown from the constructor. So the one won't be able to call `register()` for invalid collector type.

**Pull request merged**